Plutarch/Generic.hs

```diff
@@ -0,0 +1,36 @@
+{-# LANGUAGE AllowAmbiguousTypes #-}
+{-# LANGUAGE UndecidableInstances #-}
+
+-- | Common generics-sop utilities for use in Plutarch.
+module Plutarch.Generic (
```

Comment:
i don't see why this code should be exported. Can you move this into the module where you use it or name it something with `Internal` in it?

Reply:
It may come in handy for further generics deriving work (PlutusType, etc.), so I figured it is worth having a common module. Okay, I've moved it to `Plutarch.DataRepr.Internal.Generic`.